### PR TITLE
Add container-name configuration for kubernetes extension

### LINKED ISSUE #24676: kubernetes: allow to customize the container name
### Description

As today the name of the `container` that the Kubernetes generates matches the name of the `artifactId` or the value of `quarkus.kubernetes.name` if that's set however the same value is used to set i.e. the name of the generated deployment resources.

In essence it behave as:

```yaml
apiVersion: "apps/v1"
kind: "Deployment"
metadata:
  name: ${quarkus.kubernetes.name}
spec:
  template:
    spec:
      containers:
      - image: "foo/bar:latest"
        name: ${quarkus.kubernetes.name}
```

But in some cases, it is needed to set a specific name to a container and as today that's only possible by post by post processing the generated kubernetes resource yaml.

### Implementation ideas

A new property, like `quarkus.kubernetes.container-name` can be introduced to allow users to set the name of the container.
For backward compatibility, the default of the new property could be  `${quarkus.kubernetes.name}` 

```yaml
apiVersion: "apps/v1"
kind: "Deployment"
metadata:
  name: ${quarkus.kubernetes.name}
spec:
  template:
    spec:
      containers:
      - image: "foo/bar:latest"
        name: ${quarkus.kubernetes.container-name}
```

### PR DESCRIPTION
**Pull request by steel-judoka**
Resolves #24676

Add container-name configuration for kubernetes/openshift/knative to specify container name separately, backward compatible.

@geoand also after refactoring that took place in [#12655](https://github.com/quarkusio/quarkus/pull/12655) fix [#11559](https://github.com/quarkusio/quarkus/pull/11559) was lost, I added it back

### CODE REVIEW

**Comment by geoand** [comment]
Hi thanks for this.

I would prefer there was a separate commit addressing the issue that the PR is closing and another for the fix you mentioned was lost

**Comment by Yelzhasdev** [comment]
> Hi thanks for this.
> 
> I would prefer there was a separate commit addressing the issue that the PR is closing and another for the fix you mentioned was lost

Hi! Done👍

**Comment by geoand** [comment]
Thanks.

@iocanel can you please review?

**Review by iocanel — CHANGES_REQUESTED** [review]
What I see here is good.
However, I think that there is something missing.
We need to make sure that setting a custom container name does not break all other dekorators. 

We should either make force this decorator to run before all other Application Container Decorators and then pass the custom container name to the rest, all to force it last. The latter seems simpler.

However, we do need to test using the custom name with at least a couple other container related changes to make sure everything plays nicely together.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeContainerNameInDeploymentTriggerDecorator.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public class ChangeContainerNameInDeploymentTriggerDecorator extends NamedResourceDecorator<DeploymentConfigSpecFluent<?>> {
+
+    private final String containerName;
+
+    public ChangeContainerNameInDeploymentTriggerDecorator(String containerName) {
+        this.containerName = containerName;
+    }
+
+    @Override
+    public void andThenVisit(DeploymentConfigSpecFluent<?> deploymentConfigSpecFluent, ObjectMeta objectMeta) {
```
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeContainerNameInDeploymentTriggerDecorator.java:18]: Respect!

**Comment by Yelzhasdev** [comment]
> What I see here is good. However, I think that there is something missing. We need to make sure that setting a custom container name does not break all other dekorators.
> 
> We should either make force this decorator to run before all other Application Container Decorators and then pass the custom container name to the rest, all to force it last. The latter seems simpler.
> 
> However, we do need to test using the custom name with at least a couple other container related changes to make sure everything plays nicely together.

Hi! I added all decorators that extend ApplicationContainerDecorator to perform before that new decorator, also added couple tests that change container to verify that everything is correct

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeContainerNameDecorator.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        this.name = name;
+    }
+
+    @Override
+    public void andThenVisit(ContainerFluent<?> containerFluent) {
+        containerFluent.withName(name);
+    }
+
+    @Override
+    public Class<? extends Decorator>[] after() {
```
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeContainerNameDecorator.java:39]: We need to find a better way to deal with it, but there are not much we can do atm.

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**